hive/lib/src/registry/type_registry.dart

```diff
@@ -5,7 +5,7 @@ part of hive;
 /// TypeIds have to be unique and must not change.
 abstract class TypeRegistry {
   /// Register a [TypeAdapter] to announce it to Hive.
-  void registerAdapter<T>(TypeAdapter<T> adapter);
+  void registerAdapter<T>(TypeAdapter<T> adapter, {bool internal = false});
```

Comment:
I think exposing the `internal` parameter will create confusion because internal `typeIds` are treated differently (no offset) and it is very easy to mix things up. I'd rather find a way to use the matching adapter that has been registered last. Currently we always use the first matching adapter.

Reply:
Ok, for now I'm hiding `internal` parameter for now.